offchain-modules/src/packages/handlers/eos.ts

```diff
@@ -0,0 +1,212 @@
+        if (act.account !== EosTokenAccount || act.name !== EosTokenTransferActionName) {
+          continue;
+        }
+        const data = act.data;
+        if (data.to !== this.config.bridgerAccount) {
+          continue;
+        }
+        const amountAsset = parseAssetAmount(data.quantity, EosDecimal);
+        const lockEvent = {
+          TxHash: actionTrace.trx_id,
+          BlockNumber: actionTrace.block_num,
+          AccountActionSeq: action.account_action_seq,
+          GlobalActionSeq: action.global_action_seq,
+          Asset: amountAsset.Asset,
+          From: data.from,
+          To: data.to,
+          Amount: amountAsset.Amount,
+          Memo: data.memo,
+        };
+        logger.info(
+          `Eos watched transfer blockNumber:${actionTrace.block_num} accountActionSeq:${action.account_action_seq} txHash:${actionTrace.trx_id} from:${data.from} to:${data.to} amount:${lockEvent.Amount} asset:${lockEvent.Asset} memo:${data.memo}`,
+        );
+        try {
+          await this.processLockEvent(lockEvent);
+          latestActionSeq = action.account_action_seq;
+          pos -= offset + 1;
+          forward = true;
+        } catch (err) {
+          logger.error(
+            `process eosLock event failed. blockNumber:${actionTrace.block_num} tx:${lockEvent.TxHash} from:${lockEvent.From} amount:${lockEvent.Amount} asset:${lockEvent.Asset} memo:${lockEvent.Memo} error:${err}.`,
+          );
+        }
+      }
+    }
+  }
+
+  async processLockEvent(lockEvent: EosLockEvent) {
+    await this.db.createCkbMint([
+      {
+        id: lockEvent.TxHash,
+        chain: ChainType.EOS,
+        amount: lockEvent.Amount,
+        asset: lockEvent.Asset,
+        recipientLockscript: lockEvent.Memo,
+      },
+    ]);
+    await this.db.createEosLock([
+      {
+        accountActionSeq: lockEvent.AccountActionSeq,
+        globalActionSeq: lockEvent.GlobalActionSeq,
+        txHash: lockEvent.TxHash,
+        amount: lockEvent.Amount,
+        token: lockEvent.Asset,
+        sender: lockEvent.From,
+        memo: lockEvent.Memo,
+        blockNumber: lockEvent.BlockNumber,
+      },
+    ]);
+    logger.info(
+      `process CkbMint and EosLock successful for eos tx:${lockEvent.TxHash} from:${lockEvent.From} amount:${lockEvent.Amount} asset:${lockEvent.Asset} memo:${lockEvent.Memo}.`,
+    );
+  }
+
+  async watchUnlockEvents() {
+    while (true) {
+      const todoRecords = await this.getUnlockRecords('todo');
+      if (todoRecords.length === 0) {
+        await asyncSleep(15000);
+        continue;
+      }
+      await this.processUnLockEvents(todoRecords);
+    }
+  }
+
+  async processUnLockEvents(records: EosUnlock[]) {
+    for (const record of records) {
+      try {
+        record.status = 'pending';
+        await this.db.saveEosUnlock([record]);
+        const txResult = await this.sendUnlockTx(record);
+        if ('transaction_id' in txResult) {
+          record.status = 'success';
```

Comment:
The tx may be revert. Add a todo here, mark it as `pending` when send success, and `success` when confirmed.

Reply:
fixed